icu4c/source/i18n/number_longnames.cpp

```diff
@@ -448,10 +448,8 @@ void MixedUnitLongNameHandler::processQuantity(DecimalQuantity &quantity, MicroP
 const Modifier *MixedUnitLongNameHandler::getMixedUnitModifier(DecimalQuantity &quantity,
                                                                MicroProps &micros,
                                                                UErrorCode &status) const {
-    // TODO(icu-units#21): mixed units without usage() is not yet supported.
-    // That should be the only reason why this happens, so delete this whole if
-    // once fixed:
     if (micros.mixedMeasuresCount == 0) {
+        U_ASSERT(micros.mixedMeasuresCount > 0); // Mixed unit: we must have more than one unit value
```

Comment:
FYI: why I like asserts for debug builds: easier to see where they came from than to trace where status got set to UNSUPPORTED_ERROR.

Reply:
no problem